PartiQL Specification Request Proposal

dlurton opened the pull request:
The rendered markdown can be viewed [here](https://github.com/partiql/partiql-spec/blob/b2866f0851e6d3be04c08bc3d5d63cc7878eab36/psr/README-PSR.md)--useful due to flow chart.



Under this proposal, all PSRs would exist as individual files in the `psr` directory and the official spec would reside in `spec`.  I've gone ahead and arranged the directory structure to match this as part of this PR.

The complete directory structure will now be:

```
.
├── CODE_OF_CONDUCT.md
├── CONTRIBUTING.md
├── LICENSE
├── NOTICE
├── psr
│   ├── psr-process.jpeg
│   ├── PSR-TEMPLATE.md
│   └── README-PSR.md
├── README.md
└── spec
    ├── environment.tex
    ├── from.tex
    ├── groupby.tex
    ├── introduction.tex
    ├── let.tex
    ├── license.tex
    ├── main.tex
    ├── Makefile
    ├── model.tex
    ├── orderby.tex
    ├── paths.tex
    ├── pivot.tex
    ├── preamble.tex
    ├── predsFunctions.tex
    ├── schema.tex
    ├── scoping.tex
    ├── select.tex
    ├── setops.tex
    ├── subqueryCoercion.tex
    └── where.tex

```
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

almann reviewed (commented):
Conceptually, I like the idea, but what is a "feature flag?"  It isn't defined in this document, so I am not exactly clear what to make of it.

cy-partiql commented:
When implementation deviates the PSR, the next step "Revise the PSR" seems should be pointed back to "revise and review".

therapon commented on psr/README-PSR.md:
```diff
@@ -0,0 +1,46 @@
+
+
+A Proposal for Proposing Specification Changes
```
```suggestion
A Proposal for  Specification Changes
```

dlurton replied:
I was being cheeky.  :)  But yes, I will remove it.

therapon reviewed (commented):
We need a process like this. We should be a little more specific 

* What are these flags and how do we support them 
* A template issue that people can fill in with sections to guide them  
    1. Motivation 
    1. Syntax 
    1. Evalution 
    1.  Q&A

Here are some templates for ideas 

* SRFI https://srfi.schemers.org/srfi-template.html
* Rust RFC: https://github.com/rust-lang/rfcs/blob/master/0000-template.md

dlurton commented:
Also, the template I added here is simplified and modified for our purposes from the Rust equivalent: https://www.github.com/rust-lang/rfcs/tree/master/0000-template.md

Thought that might be a reasonable starting point anyway.

I am not sure if this means I have other diligence to do for anything regarding licensing...  The license of the original  appears to be Apache 2 and MIT.

RCHowell commented:
Closing in favor of the RFC
